.specify/scripts/bash/update-agent-context.sh

```
@@ -0,0 +1,781 @@
+    
+    if grep -q "^## Recent Changes" "$target_file" 2>/dev/null; then
+        has_recent_changes=1
+    fi
+    
+    # Process file line by line
+    local in_tech_section=false
+    local in_changes_section=false
+    local tech_entries_added=false
+    local changes_entries_added=false
+    local existing_changes_count=0
+    local file_ended=false
```

Shellcheck (0.11.0)

[warning] file_ended appears unused. Verify use (or export if used externally).

(SC2034)